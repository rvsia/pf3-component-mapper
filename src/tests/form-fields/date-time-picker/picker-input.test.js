import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import PickerInput from '../../../form-fields/date-time-picker/picker-input';

describe('<PickerInput />', () => {
  let initialProps;
  const handleOverlayToggle = jest.fn();
  beforeEach(() => {
    initialProps = {
      handleOverlayToggle,
    };
  });

  afterEach(() => {
    handleOverlayToggle.mockReset();
  });

  it('should render picker input correctly', () => {
    const wrapper = mount(<PickerInput { ...initialProps }/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render picker input disbaled variant correctly', () => {
    const wrapper = mount(<PickerInput { ...initialProps } isDisabled />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render picker input time variant correctly', () => {
    const wrapper = mount(<PickerInput { ...initialProps } variant="time" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should toggle overlay on input click', () => {
    const wrapper = mount(<PickerInput { ...initialProps }/>);
    wrapper.find('input').simulate('click');
    expect(handleOverlayToggle).toHaveBeenCalledWith(true);
  });

  it('should toggle overlay icon click', () => {
    const wrapper = mount(<PickerInput { ...initialProps }/>);
    wrapper.find('span.input-group-addon').simulate('click');
    expect(handleOverlayToggle).toHaveBeenCalledWith(true);
  });
});
