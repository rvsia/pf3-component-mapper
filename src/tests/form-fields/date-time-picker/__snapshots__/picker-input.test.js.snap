// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PickerInput /> should render picker input correctly 1`] = `
<PickerInput
  handleOverlayToggle={[MockFunction]}
>
  <InputGroup
    bsClass="input-group"
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <span
      className="input-group"
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <FormControl
        bsClass="form-control"
        className="picker-input"
        componentClass="input"
        onClick={[Function]}
        readOnly={true}
        style={
          Object {
            "backgroundColor": "white",
            "color": "black",
            "cursor": "pointer",
          }
        }
        type="text"
        value=""
      >
        <input
          className="picker-input form-control"
          onClick={[Function]}
          readOnly={true}
          style={
            Object {
              "backgroundColor": "white",
              "color": "black",
              "cursor": "pointer",
            }
          }
          type="text"
          value=""
        />
      </FormControl>
      <InputGroupAddon
        bsClass="input-group-addon"
        onClick={[Function]}
      >
        <span
          className="input-group-addon"
          onClick={[Function]}
        >
          <Icon
            name="calendar"
            type="fa"
          >
            <FontAwesome
              name="calendar"
            >
              <span
                aria-hidden={true}
                className="fa fa-calendar"
              />
            </FontAwesome>
          </Icon>
        </span>
      </InputGroupAddon>
    </span>
  </InputGroup>
</PickerInput>
`;

exports[`<PickerInput /> should render picker input disbaled variant correctly 1`] = `
<PickerInput
  handleOverlayToggle={[MockFunction]}
  isDisabled={true}
>
  <InputGroup
    bsClass="input-group"
    disabled={true}
    style={
      Object {
        "cursor": "not-allowed",
      }
    }
  >
    <span
      className="input-group"
      disabled={true}
      style={
        Object {
          "cursor": "not-allowed",
        }
      }
    >
      <FormControl
        bsClass="form-control"
        className="picker-input"
        componentClass="input"
        disabled={true}
        onClick={[Function]}
        readOnly={true}
        style={
          Object {
            "backgroundColor": "white",
            "color": "black",
            "cursor": "not-allowed",
          }
        }
        type="text"
        value=""
      >
        <input
          className="picker-input form-control"
          disabled={true}
          onClick={[Function]}
          readOnly={true}
          style={
            Object {
              "backgroundColor": "white",
              "color": "black",
              "cursor": "not-allowed",
            }
          }
          type="text"
          value=""
        />
      </FormControl>
      <InputGroupAddon
        bsClass="input-group-addon"
        disabled={true}
        onClick={[Function]}
      >
        <span
          className="input-group-addon"
          disabled={true}
          onClick={[Function]}
        >
          <Icon
            name="calendar"
            type="fa"
          >
            <FontAwesome
              name="calendar"
            >
              <span
                aria-hidden={true}
                className="fa fa-calendar"
              />
            </FontAwesome>
          </Icon>
        </span>
      </InputGroupAddon>
    </span>
  </InputGroup>
</PickerInput>
`;

exports[`<PickerInput /> should render picker input time variant correctly 1`] = `
<PickerInput
  handleOverlayToggle={[MockFunction]}
  variant="time"
>
  <InputGroup
    bsClass="input-group"
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <span
      className="input-group"
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <FormControl
        bsClass="form-control"
        className="picker-input"
        componentClass="input"
        onClick={[Function]}
        readOnly={true}
        style={
          Object {
            "backgroundColor": "white",
            "color": "black",
            "cursor": "pointer",
          }
        }
        type="text"
        value=""
      >
        <input
          className="picker-input form-control"
          onClick={[Function]}
          readOnly={true}
          style={
            Object {
              "backgroundColor": "white",
              "color": "black",
              "cursor": "pointer",
            }
          }
          type="text"
          value=""
        />
      </FormControl>
      <InputGroupAddon
        bsClass="input-group-addon"
        onClick={[Function]}
      >
        <span
          className="input-group-addon"
          onClick={[Function]}
        >
          <Icon
            name="time"
            type="fa"
          >
            <FontAwesome
              name="time"
            >
              <span
                aria-hidden={true}
                className="fa fa-time"
              />
            </FontAwesome>
          </Icon>
        </span>
      </InputGroupAddon>
    </span>
  </InputGroup>
</PickerInput>
`;
